### PR TITLE
Added options for filters, encapsulated WebAdmin data, fixed previewing

### PR DESCRIPTION
**Pull request by clinei**
The blog name and description can now be changed in the active configuration.
The information necessary to render WebAdmin pages is now encapsulated in structs.
Previewing now works.
Made Markdown filtering optional, but enabled by default.
Laid groundwork for adding more text filters.

Sorry for the squashed commit.

### CODE REVIEW

File: source/vibelog/web.d [hunk 1]
```diff
@@ -112,25 +116,26 @@ private final class VibeLogWeb {
...
-		c.authorIP = req.peer;
-		if (auto fip = "X-Forwarded-For" in req.headers) c.authorIP = *fip;
-		if (c.authorHomepage == "http://") c.authorHomepage = "";
-		c.content = message;
-		m_ctrl.db.addComment(post.id, c);
-
-		redirect(m_subPath ~ "posts/" ~ post.name);
+		auto post = m_ctrl.db.getPost(_slug);
+
+		if( post.commentsAllowed )
```
Comment (s-ludwig) [inline source/vibelog/web.d]: I'd prefer the old behavior or outputting an explicit error message. `@errorDisplay!getPost` and the required additions to `getPost` would then be needed to display the message properly.

File: source/vibelog/webadmin.d [hunk 2]
```diff
@@ -288,28 +365,34 @@ private final class VibeLogWebAdmin {
 			enforce(_postname == p.name, "URL does not match the edited post!");
 		} else {
 			p = new Post;
-			p.category = "default";
+			p.category = "meta";
```
Comment (s-ludwig) [inline source/vibelog/webadmin.d]: What was the reason for going with "meta" instead of "default"?

**Comment by s-ludwig** [comment]
I think the changes of how the Diet layouts/blocks are organized may need to be reverted for now. This part clearly needs some cleanup, but any changes are potentially breaking dependent code (where the blog is embedded in another site). At least I need to have a closer look at some of these projects, before I'm able to decide on a particular approach.

Otherwise - even if the commit is borderline in size ;) - looks like a very nice set of improvements and I don't have any further objections!

**Comment by lablanu** [comment]
Explicit error message implemented.
For the default category, I was going for "general".
If you `grep "block vibelog-" -rn` in the `views` directory, you'll see that the frontend blocks haven't been changed. Dependent code using `vibelog-content` and `vibelog-localnav` won't break. I'll try to get the organization back, but I doubt it will be worth it.

**Comment by s-ludwig** [comment]
My concern is that someone might have overridden `layout.dt`. But I've had a look and in my projects I only ever override `vibelog.admin.layout.dt` and `vibelog.blocks.dt`. So I don't know, maybe the risk is acceptable.

"general" sounds good.

File: views/vibelog.post.dt [hunk 3]
```diff
@@ -24,27 +24,9 @@ block vibelog-content
...
-									| anonymous
-								- if( c.authorMail.length > 0 )
-									|!= "&lt;" ~ htmlAllEscape(c.authorMail) ~ "&gt;"
-								| on #{toRFC822DateString(c.date)}, #{toRFC822TimeString(c.date)}:
-							.commentContent= c.content
-				
 				- if( info.post.commentsAllowed )
 					h2 Post a comment
-					form(action="#{info.post.name}/post_comment", method="POST")
+					form(action="#{info.refPath}/post_comment", method="POST")
```
Comment (s-ludwig) [inline views/vibelog.post.dt]: This would fail if the web interface is registered in a sub path (e.g. http://vibed.org/blog/).

**Comment by lablanu** [comment]
Latest additions include:
- Use classes instead of nested structs for data encapsulation
- Blog name and description were moved to VibeLogSettings
- Enabled using sub-paths

Data encapsulation could use `std.typecons : scoped`, I plan on adding that next.

**Comment by s-ludwig** [comment]
Hm, why did you close this? I thought that this was a great set of improvements. Unfortunately I can't always guarantee a quick response, as I simply have far too much on the table.

**Comment by clinei** [comment]
A better commit history, a recent name change and an inaccurate branch name were the main reasons. 
Also, `view` or `viewData` is a better name for the data passed to the Diet templates.  I'll clean it up and make another pull request, but before I do that, should I revert the template block layout?

**Comment by s-ludwig** [comment]
Regarding the `class`/`scoped` change, what was the reason for switching to `class`? The `structs` should be zero-cost, because they are passed to `render` using an `alias` template parameter, which means that `render` basically is a delegate that has access to the original instances within the enclosing stack frame. So in that regard, using a class would just produce a bit more syntax overhead.

The `info` -> `view` change should ideally be a separate PR, I don't feel strongly about the name, but such big-diff-small-change change sets ideally shouldn't be mixed with functional changes to ease the review.

I'd say, keep the template block layout changes. The next release will have to bump the minor version number anyway, so that this can as well be cleaned up properly. Maybe I'll in fact do some further changes later on.

**Comment by clinei** [comment]
Reverted to `struct` with `alias`.

For the data passed to `render`, what about `viewData`? It is more accurate than `info`/`view`, if a bit more verbose.

**Pull request closed without merging**